src/frontend/admin/src/hooks/useOrders/useOrders.tsx

```diff
@@ -85,6 +94,36 @@ const orderProps: UseResourcesProps<Order, OrderQuery> = {
 };
 
 // eslint-disable-next-line react-hooks/rules-of-hooks
-export const useOrders = useResources(listProps);
+export const useOrders = (
+  filters?: OrderQuery,
+  queryOptions?: QueryOptions<OrderListItem>,
+) => {
+  const intl = useIntl();
+  const custom = useResourcesCustom({ ...listProps, filters, queryOptions });
+  const mutation = useMutation;
+  return {
+    ...custom,
+    methods: {
+      ...custom.methods,
+      generateCertificate: mutation({
+        mutationFn: async (data: { orderId: string }) => {
+          return OrderRepository.generateCertificate(data.orderId);
+        },
+        onSuccess: () => {
+          custom.methods.invalidate();
+          custom.methods.showSuccessMessage(
+            intl.formatMessage(useOrdersMessages.successCertificateGenerate),
+          );
+        },
+        onError: (error: any) => {
+          custom.methods.setError(
+            error?.data?.details ??
+              intl.formatMessage(useOrdersMessages.errorUpdate),
+          );
+        },
```

Comment:
The method `checkStatus` return a `HttpError` in case of error, so let's use the right type here.
```suggestion
        onError: (error: HttpError) => {
          custom.methods.setError(
            error.data?.details ??
              intl.formatMessage(useOrdersMessages.errorUpdate),
          );
        },
```